packages/vm/tests/api/state/stateManager.spec.ts

```diff
@@ -409,3 +409,80 @@ tape('StateManager - Contract code', (tester) => {
     t.end()
   })
 })
+
+tape('StateManager - Contract storage', (tester) => {
+  const it = tester.test
+
+  it('should throw on storage values larger than 32 bytes', async (t) => {
+    t.plan(1)
+    const stateManager = new DefaultStateManager()
+    const address = zeros(20)
+    const key = zeros(32)
+    const value = Buffer.from('aa'.repeat(33), 'hex')
+    try {
+      await stateManager.putContractStorage(address, key, value)
+      t.fail('did not throw')
+    } catch (e) {
+      t.pass('threw on trying to set storage values larger than 32 bytes')
+    }
+    t.end()
+  })
+
+  it('should strip storage values lower than 32 bytes', async (t) => {
+    const stateManager = new DefaultStateManager()
+    const address = zeros(20)
+    const key0 = zeros(32)
+    const value0 = Buffer.from('aa'.repeat(31), 'hex') // put a value of 31-bytes length
+    const expect0 = unpadBuffer(value0)
+    const key1 = Buffer.concat([zeros(31), Buffer.from('01', 'hex')])
+    const value1 = Buffer.from('aa'.repeat(1), 'hex') // put a vlaue of 1-byte length
+    const expect1 = unpadBuffer(value1)
+
+    await stateManager.putContractStorage(address, key0, value0)
+    const slot0 = await stateManager.getContractStorage(address, key0)
+    t.ok(slot0.equals(expect0), 'value of 31 bytes padded correctly')
```

Comment:
Really nit, but I would think this test would be a bit better readable if the respective execution part of these two cases are combined with the value setup (so everything with *0 together and everything with *1 together)

Reply:
Fixed -> 5217f50

Note: I have edited this test a little bit (it did `unpadZeros` in previous commits on this particular test, but none of the test cases actually had leading zeros)